codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGeneratorV2.kt

```diff
@@ -469,8 +472,143 @@ class ServerServiceGeneratorV2(
+            ///
+            /// #### Plugins
+            ///
+            /// The [`$serviceName::builder_with_plugins`] method, returning [`$builderName`],
+            /// accepts a [`Plugin`](aws_smithy_http_server::plugin::Plugin).
+            /// Plugins allow you to build middleware which is aware of the operation it is being applied to.
+            ///
+            /// ```rust,ignore
+            /// ## use #{SmithyHttpServer}::plugin::IdentityPlugin as LoggingPlugin;
+            /// ## use #{SmithyHttpServer}::plugin::IdentityPlugin as MetricsPlugin;
+            /// ## use #{SmithyHttpServer}::plugin::PluginPipeline;
+            /// let plugins = PluginPipeline::new()
+            ///         .push(LoggingPlugin)
+            ///         .push(MetricsPlugin);
+            /// let builder = $crateName::$serviceName::builder_with_plugins(plugins);
+            /// ```
+            ///
+            /// To know more about plugins, see: [`#{SmithyHttpServer}::plugin`].
+            ///
+            /// #### Handlers
+            ///
+            /// For each operation [`$builderName`] has an associated setter accepting an async function conforming to the Smithy model.
+            /// We call these async functions **handlers**. This is where your application business logic lives.
+            ///
+            /// Every handler must take an `Input`, and optional [`extractor arguments`](#{SmithyHttpServer}::request), while returning:
+            ///
+            /// * A `Result<Output, Error>` if your operation has modeled errors, or
+            /// * An `Output` otherwise.
+            ///
+            /// ```rust,ignore
+            /// async fn fallible_handler(input: Input, extensions: #{SmithyHttpServer}::Extension<T>) -> Result<Output, Error> { todo!() }
+            /// async fn infallible_handler(input: Input, extensions: #{SmithyHttpServer}::Extension<T>) -> Output { todo!() }
+            /// ```
+            ///
+            /// Handlers accept up to 8 extractors:
+            ///
+            /// ```rust,ignore
+            /// async fn handler_with_no_extensions(input: Input) -> ... { todo!() }
+            /// async fn handler_with_one_extension(input: Input, ext: #{SmithyHttpServer}::Extension<T>) -> ... { todo!() }
+            /// async fn handler_with_two_extensions(input: Input, ext0: #{SmithyHttpServer}::Extension<T>, ext1: #{SmithyHttpServer}::Extension<T>) -> ... { todo!() }
+            /// ...
+            /// ```
+            ///
+            /// #### Build
+            ///
+            /// When you have set all the operations you want to serve, you can construct [`$serviceName`] using:
+            ///
+            /// * [`$builderName::build`]: a fallible constructor, returning an error describing the missing operations from all possible ones in the Smithy model
+            /// * [`$builderName::build_unchecked`]: an infallible constructor, returning a 500 when requesting missing, unset operations
```

Comment:
```suggestion
            /// You can convert [`$builderName`] into [`$serviceName`] using either [`$builderName::build`] or [`$builderName::build_unchecked`].  
            ///
            /// [`$builderName::build`] requires you to provide a handler for every single operation in your Smithy model. It will return an error if that is not the case.  
            ///
            /// [`$builderName::build_unchecked`], instead, does not require exhaustiveness. The server will automatically return 500s to all requests for operations that do not have a registered handler.  
            /// [`$builderName::build_unchecked`] is particularly useful if you are deploying your Smithy service as a collection of Lambda functions, where each Lambda is only responsible for a subset of the operations in the Smithy service (or even a single one!).
```